common/app/com/commercetools/sunrise/framework/controllers/SunriseController.java

```diff
@@ -22,4 +20,26 @@ public final Logger getLogger() {
     protected final CompletionStage<Result> redirectToCall(final Call call) {
         return completedFuture(Results.redirect(call));
     }
+
+    /**
+     * Saves a message of a certain {@link MessageType} meant to be displayed to the user after a redirection to
+     * another page.
+     *
+     * Example use case: the user submitted a successful form, then the user is redirected to a page where the message
+     * informing of the successful action is shown.
+     * In this case {@code saveMessage(MessageType.SUCCESS, "The action was successful!")} would be invoked right
+     * before the redirection is executed.
+     *
+     * @param messageType type of the message to be saved (e.g. success, info)
+     * @param message the content of the message to be saved
+     */
+    protected void saveMessage(final MessageType messageType, final String message) {
```

Comment:
I think that this wouldn't allow us to send multipe messages of the same type,
I'm i correct?

Reply:
Yep, you are correct. That's a limitation of Play itself because flash doesn't allow to save objects (as it's a cookie it's also not recommended). But there are workarounds for that:
1) set in flash your own message type and extract it into view models (just around 4 lines of simple code)
2) have into the same message type multiple messages (the approach might change depending on the situation)
3) we can also easily extend the system by using the message types as prefixes and allow multiple messages without breaking changes

(Just a summary of what we talked about to have it written 😉 )